compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -916,10 +918,14 @@ parserInfo numProcessors =
         ])
     )
 
-cliArgsFromDamlYaml :: IO [String]
-cliArgsFromDamlYaml = do
+cliArgsFromDamlYaml :: Maybe ProjectOpts -> IO [String]
+cliArgsFromDamlYaml mbProjectOpts = do
+    -- This is the same logic used in withProjectRoot but we don’t need to change CWD here
+    -- and this is simple enough so we inline it here.
+    envProjectPath <- fmap ProjectPath <$> getProjectPath
+    let projectPath = fromMaybe (ProjectPath ".") ((projectRoot =<< mbProjectOpts) <|> envProjectPath)
```

Comment:
Maybe call `envProjectPath` `mbEnvProjectPath`, and name `projectRoot =<< ...` as `mbOptProjectPath` to make this a little more obvious to read?

Reply:
good idea! changed